### PR TITLE
added examples from pulumiUP

### LINKED ISSUE #476: Add demo code to examples
- https://github.com/pulumi/pulumi-java/tree/dixler/demo-4/tests/examples/eks-minimal
- https://github.com/pulumi/pulumi-java/tree/dixler/demo-4/tests/examples/aws-java-spring-eks

### PR DESCRIPTION
**Pull request by dixler**
# Description

Fixes #476 

## Checklist

- [ ] I have added tests that prove my fix is effective or that my feature works

- [ ] I have updated the [CHANGELOG-PENDING](https://github.com/pulumi/pulumi/blob/master/CHANGELOG_PENDING.md) file with my change

- [ ] Yes, there are changes in this PR that warrants bumping the Pulumi Service API version

### CODE REVIEW

File: tests/examples/aws-java-pulumiup-app/pom.xml [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+            <version>4.37.3</version>
+        </dependency>
+
+        <dependency>
+            <groupId>com.pulumi</groupId>
+            <artifactId>kubernetes</artifactId>
+            <version>3.15.1</version>
+        </dependency>
+
+		<dependency>
```
Comment (pawelprazak) [inline tests/examples/aws-java-pulumiup-app/pom.xml:49]: nit: formatting issue, tabs vs spaces?

File: tests/examples/aws-java-pulumiup-app/src/main/java/com/example/pulumi/JarDeployment.java [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+        // Check out JarDeploymentArgs
+        final var args = JarDeployment.apply(applyArgs);
+
+        final var appJar = deployJar(args.getJarPath(), args.getBucketName());
+        this.endpoint = deployApp(args.getKubeconfig(), appJar);
+
+        return;
+    }
+
+    public interface LambdaArgs {
```
Comment (pawelprazak) [inline tests/examples/aws-java-pulumiup-app/src/main/java/com/example/pulumi/JarDeployment.java:54]: nit: you can add `@FunctionalInterface` https://docs.oracle.com/javase/8/docs/api/java/lang/FunctionalInterface.html

File: tests/examples/aws-java-pulumiup-platform/app/src/main/java/com/broom/example/EKSStack.java [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+public class EKSStack {
+
+    private final Cluster cluster;
+
+    /**
+     * Creates an EKS cluster
+     * @param clusterPolicies
+     */
+    public EKSStack(List<Policy> clusterPolicies) {
```
Comment (pawelprazak) [inline tests/examples/aws-java-pulumiup-platform/app/src/main/java/com/broom/example/EKSStack.java:27]: nit: could have varargs overload

File: tests/examples/aws-java-pulumiup-app/src/test/java/com/example/springboot/HelloControllerIT.java [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+import org.springframework.boot.test.web.client.TestRestTemplate;
+import org.springframework.http.ResponseEntity;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+public class HelloControllerIT {
+
+	@Autowired
+	private TestRestTemplate template;
```
Comment (pawelprazak) [inline tests/examples/aws-java-pulumiup-app/src/test/java/com/example/springboot/HelloControllerIT.java:16]: formating issue

File: tests/examples/aws-java-pulumiup-app/src/main/java/com/example/pulumi/PulumiStack.java [hunk 5]
```diff
@@ -0,0 +1,36 @@
+package com.example.pulumi;
+
+import com.pulumi.Context;
+import com.pulumi.Pulumi;
+import com.pulumi.core.Output;
+import com.pulumi.resources.StackReference;
+
+public class PulumiStack {
+
+	public static void main(String[] args) {
```
Comment (pawelprazak) [inline tests/examples/aws-java-pulumiup-app/src/main/java/com/example/pulumi/PulumiStack.java:10]: formatting issue

**Review by pawelprazak — APPROVED** [review]
Looks good, just a couple of hints, and also changelog is missing.

File: tests/examples/aws-java-pulumiup-app/src/test/java/com/example/springboot/HelloControllerIT.java (hunk 4 above)
Comment (dixler) [reply tests/examples/aws-java-pulumiup-app/src/test/java/com/example/springboot/HelloControllerIT.java:16]: I copy pasted this. I'm going to add a readme with attribution

**Pull request merged**